src/object/cli_obj.c

```diff
@@ -7270,8 +7268,13 @@ dc_obj_sync(tse_task_t *task)
 		D_ASSERTF(*args->nr == obj->cob_grp_nr, "Invalid obj sync args %d/%d\n",
 			  *args->nr, obj->cob_grp_nr);
 
-		for (i = 0; i < *args->nr; i++)
-			*args->epochs_p[i] = 0;
+		if (obj_auxi->args_initialized) {
+			/* Remove previous shard tasks, since related leader(s) maybe changed. */
+			obj_io_set_new_shard_task(obj_auxi);
+			obj_auxi->args_initialized = 0;
+		}
+
+		memset(*args->epochs_p, 0, sizeof(daos_epoch_t) * *args->nr);
```

Comment:
is it possible the "*args->epochs_p" is NULL?

Reply:
It will not, there is check "D_ASSERT(*args->epochs_p != NULL);" at L7267.